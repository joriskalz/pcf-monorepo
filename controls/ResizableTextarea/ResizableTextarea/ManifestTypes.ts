export enum BorderStyles {
  Normal = 0,
  Centred = 1,
}

export const ResizeTypes: Record<"0" | "1" | "2" | "3", string> = {
  "0": "none",
  "1": "both",
  "2": "vertical",
  "3": "horizontal",
};

export interface InputParameters {
  DefaultHeight?: number | null;
  DefaultWidth?: number | null;
  InputEvent?: string | null;
  Text: string;
  Default: string;
}

export interface OutputParameterFlags {
  resizedHeight?: boolean;
  resizedWidth?: boolean;
  text?: boolean;
}

export const enum InputEvents {
  SetFocus = "SetFocus",
  ResetSize = "ResetSize",
  ResetWidth = "ResetWidth",
  ResetHeight = "ResetHeight",
  ResetText = "ResetText",
}

export const enum FontSizeUnits {
  Pt = "0",
  Px = "1",
}
