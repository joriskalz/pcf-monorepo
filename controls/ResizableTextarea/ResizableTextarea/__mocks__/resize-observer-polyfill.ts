/* istanbul ignore file */

/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
/* eslint-disable @typescript-eslint/no-unused-vars */

export const ResizeObserverMock = {
  onResize: jest.fn(),
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
};
export default class ResizeObserver {
  constructor(handler: any) {
    ResizeObserverMock.onResize = handler;
    this.observe = ResizeObserverMock.observe;
    this.unobserve = ResizeObserverMock.unobserve;
    this.disconnect = ResizeObserverMock.disconnect;
  }
  disconnect(): void;
  disconnect(): void;
  disconnect(): void {
    throw new Error();
  }
  observe(target: Element, options?: ResizeObserverOptions): void;
  observe(target: Element): void;
  observe(target: any, options?: any): void {
    throw new Error();
  }
  unobserve(target: Element): void;
  unobserve(target: Element): void;
  unobserve(target: any): void {
    throw new Error();
  }
}
