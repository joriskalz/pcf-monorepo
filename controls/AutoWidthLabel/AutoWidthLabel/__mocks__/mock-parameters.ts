/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
