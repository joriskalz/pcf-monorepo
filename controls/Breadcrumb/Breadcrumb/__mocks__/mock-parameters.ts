/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
