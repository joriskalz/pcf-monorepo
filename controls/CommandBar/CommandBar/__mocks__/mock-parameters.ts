/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
