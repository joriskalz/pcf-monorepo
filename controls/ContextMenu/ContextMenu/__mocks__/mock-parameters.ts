/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
