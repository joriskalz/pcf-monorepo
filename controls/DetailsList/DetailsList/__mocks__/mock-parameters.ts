/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
