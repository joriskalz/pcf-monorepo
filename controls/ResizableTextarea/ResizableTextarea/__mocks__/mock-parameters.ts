/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    Default: new MockStringProperty(),
    MaxLength: new MockWholeNumberProperty(),
    HintText: new MockStringProperty(),
    EmptyPlaceholderText: new MockStringProperty(),
    ResizedHeight: new MockWholeNumberProperty(),
    ResizedWidth: new MockWholeNumberProperty(),
    DefaultHeight: new MockWholeNumberProperty(),
    DefaultWidth: new MockWholeNumberProperty(),
    MinHeight: new MockWholeNumberProperty(),
    MaxHeight: new MockWholeNumberProperty(),
    MinWidth: new MockWholeNumberProperty(),
    MaxWidth: new MockWholeNumberProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    LeftPaddingAdjustment: new MockWholeNumberProperty(),
    AllowResize: new MockEnumProperty<"0" | "1" | "2" | "3">(),
    EnableSpellCheck: new MockEnumProperty<"0" | "1">(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty(),
    FontColor: new MockStringProperty("1"),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    HintFontColor: new MockStringProperty(),
    HintFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
    AccessibilityLabel: new MockStringProperty(),
    InputEvent: new MockStringProperty(),
  };
}
