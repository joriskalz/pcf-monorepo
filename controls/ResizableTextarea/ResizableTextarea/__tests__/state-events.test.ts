import { ResizableTextarea } from "..";
import { IInputs } from "../generated/ManifestTypes";
import { MockContext, MockState } from "../__mocks__/mock-context";
import { getMockParameters } from "../__mocks__/mock-parameters";

describe("ResizeableTextArea element state events", () => {
  beforeEach(() => jest.clearAllMocks());

  it("updates states on hover and focus", () => {
    const { component, context, notifyOutputChanged, state, container } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (context as any).accessibility.assignedTabIndex = 10;
    getClassicStyles(context);

    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot();

    // Set mouse over
    const textarea = container.getElementsByTagName("textarea")[0];
    textarea.dispatchEvent(new MouseEvent("mouseenter"));
    expect(container).toMatchSnapshot("hover");

    // Set mouse out
    textarea.dispatchEvent(new MouseEvent("mouseleave"));
    expect(container).toMatchSnapshot("hover:leave");

    // Set focus
    textarea.dispatchEvent(new MouseEvent("focus"));
    expect(container).toMatchSnapshot("focus");

    // Set lose focus
    textarea.dispatchEvent(new MouseEvent("blur"));
    expect(container).toMatchSnapshot("blur");
  });

  it("sets disabled styles", () => {
    const { context, container, component, notifyOutputChanged, state } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    context.mode.isControlDisabled = true;
    getClassicStyles(context);
    // The DisabledHoverFillColor is used for MDA apps when the form is read-only and mouse over the control
    // should show a grey box to indicat it is disabled
    context.parameters.DisabledHoverFillColor.raw = "silver";
    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot("disabled");

    // Set mouse over
    const textarea = container.getElementsByTagName("textarea")[0];
    textarea.dispatchEvent(new MouseEvent("mouseenter"));
    expect(container).toMatchSnapshot("hover");

    // Set mouse out
    textarea.dispatchEvent(new MouseEvent("mouseleave"));
    expect(container).toMatchSnapshot("hover:leave");
  });

  it("sets read-only styles due to security", () => {
    const { context, container, component, notifyOutputChanged, state } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    context.parameters.Text.security = {
      editable: false,
      readable: true,
      secured: true,
    };
    getClassicStyles(context);
    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot("disabled");
  });

  it("sets read-only and masked styles due to security", () => {
    const { context, container, component, notifyOutputChanged, state } = createComponent();
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    context.parameters.Text.security = {
      editable: false,
      readable: false,
      secured: true,
    };
    getClassicStyles(context);
    component.init(context, notifyOutputChanged, state, container);
    expect(container).toMatchSnapshot("disabled");
  });
});

function getClassicStyles(context: MockContext<IInputs>) {
  const { parameters } = context;
  parameters.MaxLength.raw = 100;
  parameters.HintText.raw = "Enter text here...";
  parameters.PaddingLeft.raw = 12;
  parameters.PaddingRight.raw = 5;
  parameters.PaddingTop.raw = 5;
  parameters.PaddingBottom.raw = 5;
  parameters.EnableSpellCheck.raw = "1";
  parameters.FontSize.raw = 17.5;
  parameters.FontColor.raw = "rgb(0,0,0)";
  parameters.FontWeight.raw = "400";
  parameters.DisabledFontColor.raw = "rgb(166,166,166)";
  parameters.FillColor.raw = "#ffffff";
  parameters.DisabledFillColor.raw = "rgb(244,244,244)";
  parameters.HoverFillColor.raw = "#BACAE2";
  parameters.BorderColor.raw = "#00126B";
  parameters.BorderThickness.raw = 2;
  parameters.BorderRadius.raw = 5;
  parameters.DisabledBorderColor.raw = "rgb(166,166,166)";
  parameters.FocusBorderThickness.raw = 4;
  parameters.RenderBorderStyle.raw = "1"; // Centered
  parameters.AccessibilityLabel.raw = "A textarea";
}

function createComponent() {
  const component = new ResizableTextarea();
  const notifyOutputChanged = jest.fn();
  const context = new MockContext<IInputs>(getMockParameters());
  const state = new MockState();
  const container = document.createElement("div");
  document.body.appendChild(container);
  return { component, context, notifyOutputChanged, state, container };
}
