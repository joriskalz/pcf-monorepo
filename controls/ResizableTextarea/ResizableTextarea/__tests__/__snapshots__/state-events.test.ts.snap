// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizeableTextArea element state events sets disabled styles: disabled 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: rgb(166,166,166); background: rgb(244, 244, 244);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          disabled=""
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(166, 166, 166); font-weight: 400;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events sets disabled styles: hover 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: rgb(166,166,166); background: silver;"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          disabled=""
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(166, 166, 166); font-weight: 400;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events sets disabled styles: hover:leave 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: rgb(166,166,166); background: rgb(244, 244, 244);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          disabled=""
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(166, 166, 166); font-weight: 400;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events sets read-only and masked styles due to security: disabled 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: rgb(166,166,166); background: rgb(244, 244, 244);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          disabled=""
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(166, 166, 166); font-weight: 400;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events sets read-only styles due to security: disabled 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: rgb(166,166,166); background: rgb(244, 244, 244);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          disabled=""
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(166, 166, 166); font-weight: 400;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: #00126b; background: rgb(255, 255, 255);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(0, 0, 0); font-weight: 400;"
          tabindex="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: blur 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: #00126b; background: rgb(255, 255, 255);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(0, 0, 0); font-weight: 400;"
          tabindex="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: focus 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 4px; margin: -2px; border-color: #00126b; background: rgb(255, 255, 255);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -2px;"
      >
        <textarea
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(0, 0, 0); font-weight: 400;"
          tabindex="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: hover 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: #00126b; background: rgb(186, 202, 226);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(0, 0, 0); font-weight: 400;"
          tabindex="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResizeableTextArea element state events updates states on hover and focus: hover:leave 1`] = `
<div>
  <div
    class="cat-textarea-outer"
  >
    <div
      class="cat-textarea-border-container"
      style="border-radius: 5px; border-width: 2px; margin: -1px; border-color: #00126b; background: rgb(255, 255, 255);"
    >
      <div
        class="cat-textarea-border-inner"
        style="margin: -1px;"
      >
        <textarea
          maxlength="100"
          placeholder="Enter text here..."
          style="font-size: 17.5px; padding: 5px 5px 5px 12px; resize: both; color: rgb(0, 0, 0); font-weight: 400;"
          tabindex="10"
        />
      </div>
    </div>
  </div>
</div>
`;
